### PR TITLE
Delete location before contact delete 9ci/domain9#477

### PR DESCRIPTION
**Pull request by snimavat**
connect 9ci/domain9#477

### CODE REVIEW

**Comment by snimavat** [comment]
@basejump ContactSource gets deleted as cascade delete as per mapping - test verifies it.
I have updated to remove contact location.

We have this `Contact.groovy`
```
        flex cascade: "all"
        emails cascade: "all-delete-orphan"
        phones cascade: "all-delete-orphan"
        sources cascade: "all-delete-orphan"
```

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy [hunk 1]
```diff
@@ -64,14 +64,7 @@ class ContactRepo extends LongIdGormRepo<Contact> {
...
         TagLink.remove(contact)
 
-        // XXX why are we keeping the locations around?
-        // if its a location for a contact it should be deleted along with the contact right?
-        Location.executeUpdate("update Location set contact = null where contact = :contact", [contact: contact]) //set contact to null
-
-        // XXX we are not deleting Location or CSource? Why
-        // something like this should be run no?
-        // Location.query(contact: contact).deleteAll()
-        // ContactSource.query(contact: contact).deleteAll()
```
Comment (basejump) [inline rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy:74]: what happened to ContactSource?

File: rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy [hunk 2]
```diff
@@ -64,14 +64,7 @@ class ContactRepo extends LongIdGormRepo<Contact> {
         //remove
         TagLink.remove(contact)
 
-        // XXX why are we keeping the locations around?
-        // if its a location for a contact it should be deleted along with the contact right?
-        Location.executeUpdate("update Location set contact = null where contact = :contact", [contact: contact]) //set contact to null
```
Comment (basejump) [inline rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy:69]: @jdabal this was done like this another CED hack. can you check into it. was some kind of special contact for bill to or something that had the location assigned. I looked into some months back

**Review by basejump — CHANGES_REQUESTED** [review]

**Comment by snimavat** [comment]
New PR here - https://github.com/yakworks/gorm-tools/pull/599

**Comment by snimavat** [comment]
Done here - https://github.com/yakworks/gorm-tools/pull/599

**Pull request closed without merging**